edward/inferences.py

```diff
@@ -180,16 +186,24 @@ def initialize(self, n_iter=1000, n_data=None, n_print=100,
         self.n_print = n_print
         self.loss = tf.constant(0.0)
 
-        # Set shape of data placeholders according to batch size.
-        if not isinstance(self.model, StanModel):
-            for value in six.itervalues(self.data):
-                value.set_shape([self.n_data] + get_dims(value)[1:])
+        if n_data is not None and not isinstance(self.model, StanModel):
+            # Re-assign data to batch tensors, with size given by `n_data`.
+            values = list(six.itervalues(self.data))
+            slices = tf.train.slice_input_producer(values)
+            batches = tf.train.batch(slices, self.n_data)
```

Comment:
@altosaar mentioned via Slack that we should default to a larger number of threads than 1.